src/core_tests/workflow_tasks.rs

```diff
@@ -1665,3 +1665,72 @@ async fn failing_wft_doesnt_eat_permit_forever() {
 
     core.shutdown().await;
 }
+
+#[tokio::test]
+async fn cache_miss_doesnt_eat_permit_forever() {
```

Comment:
Wouldn't it have been more useful to have an integration test here so we could verify the SDK behaves correctly with an actual server implementation?

Reply:
Added one as well